src/ansys/pytwin/evaluate/twin_model.py

```diff
@@ -1804,6 +1853,87 @@ def generate_points(self, rom_name: str, on_disk: bool = True, named_selection:
+        """
+        Project the field ROM data onto a mesh
+
+        Parameters
+        ----------
+        rom_name : str
+            Name of the TBROM considered to project the results.
+        mesh: pyvista.DataSet
+            PyVista DataSet object of the targeted mesh.
+        interpolate: bool
+            Flag to indicate whether the point cloud data are interpolated (True) or not (False) on the targeted mesh.
+            Interpolation is recommended when point cloud data and mesh data are not ordered in the same way, and when
+            the target mesh is different from the one used to generate the ROM. Interpolation is automatically enforced
+            if the target mesh size (i.e. number of cells/points) is different from the point cloud size.
+        named_selection: str (optional)
+            Named selection on which the mesh projection has to be performed. The default is ``None``, in which case the
+            entire domain is considered.
+
+        Returns
+        -------
+        pyvista.DataSet
+            PyVista DataSet object of the targeted mesh with projected field ROM data.
+
+        Raises
+        ------
+        TwinModelError:
+            If ``TwinModel`` object has not been initialized.
+            If rom_name is not included in the Twin's list of TBROM
+            If TBROM hasn't its mode coefficients outputs connected to the twin's outputs
+            If mesh is not a valid grid dataset
+            If name_selection is not included in the TBROM's list of Named Selections
+            If interpolate is True and no points file is available with the TBROM
+
+        TwinModelWarning:
+            If interpolate is False and the targeted mesh has a number of cells and points different from TBROM point
+            cloud. In that case, interpolate is automatically switched to True.
+
+        Examples
+        --------
+        >>> from pytwin import TwinModel
+        >>> import pyvista as pv
+        >>> # Instantiate a twin model, initialize it, and evaluate it step by step until you want to save its state
+        >>> model1 = TwinModel('model.twin')
+        >>> model1.initialize_evaluation()
+        >>> romname = model1.tbrom_names[0]
+        >>> mesh = pv.read('mesh.vtk')
+        >>> fieldresults = model1.project_tbrom_on_mesh(romname, mesh)
+        """
+        self._log_key = "MeshProjection"
+
+        if not self.evaluation_is_initialized:
+            msg = self._error_msg_for_not_initialized()
+            self._raise_error(msg)
+
+        try:
+            if self._check_tbrom_mesh_projection_args(rom_name, mesh, named_selection):
+                if named_selection is None:
+                    nb_points = self._tbroms[rom_name].nb_points
+                else:
+                    nb_points = len(self._tbroms[rom_name].named_selection_indexes(named_selection))
+                if not interpolate and (mesh.n_cells != nb_points and mesh.n_points != nb_points):
```

Comment:
if condition from line 1917 is quite hard to understand. I would suggest to split it into multiple levels. Also, I'm not sure the check on `mesh.n_cells != nb_points` makes sense because the pyvista interpolate method only considers the points of the target mesh to perform the interpolation.

Reply:
as discussed today : we need to test more workflows and collect more feedbacks on possible limitations